hadoop-hdds/common/src/main/java/org/apache/hadoop/util/MetricUtil.java

```diff
@@ -18,16 +18,29 @@
 package org.apache.hadoop.util;
 
 import org.apache.hadoop.metrics2.lib.MutableRate;
+import org.apache.hadoop.ozone.util.ShutdownHookManager;
 import org.apache.ratis.util.function.CheckedRunnable;
 import org.apache.ratis.util.function.CheckedSupplier;
 
 import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.function.Consumer;
 
+import static org.apache.hadoop.ozone.conf.OzoneServiceConfig.DEFAULT_SHUTDOWN_HOOK_PRIORITY;
+
 /**
  * Encloses helpers to deal with metrics.
  */
 public final class MetricUtil {
+
+  private static ExecutorService executor = Executors.newFixedThreadPool(10);
```

Comment:
Please set some thread name (`new ThreadFactoryBuilder().setNameFormat("...")`).

Nit: `executor` can be final (also needs to be renamed to `EXECUTOR` to match conventions).

Reply:
done